sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/models/traits/HttpTrait.java

```diff
@@ -115,11 +117,28 @@ public interface HttpTrait<T extends HttpTrait<T>> {
      * trait that are also ignored if an {@link HttpPipeline} is specified, so please be sure to refer to the
      * documentation of types that implement this trait to understand the full set of implications.</p>
      *
-     * @param logOptions The {@link HttpLogOptions logging configuration} to use when sending and
-     * receiving requests to and from the service.
+     * @param logOptions The {@link HttpLogOptions logging configuration} to use when sending and receiving requests to
+     * and from the service.
      *
      * @return Returns the same concrete type with the appropriate properties updated, to allow for fluent chaining of
      * operations.
      */
     T httpLogOptions(HttpLogOptions logOptions);
+
+    /**
+     * Sets the {@link HttpRedirectOptions} for all the requests made through the client.
+     *
+     * <p><strong>Note:</strong> It is important to understand the precedence order of the {@link HttpTrait} APIs. In
+     * particular, if a {@link HttpPipeline} is specified, this takes precedence over all other APIs in the trait, and
+     * they will be ignored. If no {@link HttpPipeline} is specified, an HTTP pipeline will be constructed internally
+     * based on the settings provided to this trait. Additionally, there may be other APIs in types that implement this
+     * trait that are also ignored if an {@link HttpPipeline} is specified, so please be sure to refer to the
+     * documentation of types that implement this trait to understand the full set of implications.</p>
+     *
+     * @param httpRedirectOptions The {@link HttpRedirectOptions} to use for all the requests made through the client.
+     *
+     * @return Returns the same concrete type with the appropriate properties updated, to allow for fluent chaining of
+     * operations.
+     */
+    T httpRedirectOptions(HttpRedirectOptions httpRedirectOptions);
```

Comment:
This is something we should discuss adding to azure-core as well. Redirect options are generally not needed on every client builder. So, I am inclined to not add this in the trait interface and let users use the addPolicy method to add the redirect policy.